goblint_runner.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+import argparse
+from dataclasses import dataclass
+
+import subprocess
+from os import path
+import logging
+
+
+@dataclass
+class ExecutionResult:
+    output: str
+    has_verdict: bool
+
+
+class GoblintMultishotRunner:
+
+    def __init__(self, logger):
+        self.logger = logger
+        self.current_path = path.dirname(__file__)
+        self.goblint_executable_path = path.join(self.current_path, "goblint")
+        if not path.exists(self.goblint_executable_path):
+            logger.error(f" Could not find goblint executable at {self.goblint_executable_path}; did you build goblint with make?")
+            exit(1)
+
+        parser = argparse.ArgumentParser(
+            description="""A facade in front of goblint to enable running a portfolio of configurations for SV-COMP.
+            All args apart from --portfolio-conf/-p are passed on to the actual goblint calls.
+            The portfolio config file is a plaintext file whose lines each consist of a path to a
+            goblint config file (relative to the goblint conf dir). 
+            Goblint is run with each config in order until one produces a verdict true or reaches the end of the list.
+            You may add comments to the portfolio config file by starting a line with #.
+            """
+        )
+        parser.add_argument("-p","--portfolio-conf", type=str, metavar="FILE",dest="portfolio",
+                            help="a path to a portfolio configuration file (relative to goblint_multishot.py)")
+        conf_args, self.other_args = parser.parse_known_args()
+        logger.debug(f"Portfolio-conf file: {conf_args.portfolio}")
+        logger.debug(f"Arguments passed on to goblint: {" ".join(self.other_args)}")
+
+        self.configs = []
+        if conf_args.portfolio:
+            if not path.exists(conf_args.portfolio):
+                logger.error(f" Could not find portfolio conf file at {conf_args.portfolio}")
+                exit(1)
+            with open(conf_args.portfolio, "r") as conflist_file:
+                self.configs = [path.join(self.current_path, "conf", c.strip())
+                                for c in conflist_file.readlines() if not c.strip().startswith("#")]
+            logger.info(f"Loaded goblint configs: {", ".join(self.configs)}")
+            
+    def run_with_config(self, config_path):
+        args = ["--conf", config_path] + self.other_args
+        self.logger.info(f"Running next shot: ./goblint {" ".join(args)}")
+        output = subprocess.check_output([self.goblint_executable_path, *args]
+                                        ).decode("utf-8")        
+
+        return ExecutionResult(output, self.has_verdict(output))
+
+    def run_without_config(self):
+        subprocess.run([self.goblint_executable_path, *self.other_args])
+
+    @staticmethod
+    def has_verdict(output):
+        return any("SV-COMP result: true" in line for line in output.splitlines())
```

Comment:
Currently if Goblint returns an `ERROR (something)` result, this would just cause the next conf to be run, right?

I think it would be better if such errors also immediately stopped the portfolio with that result. So I guess only `SV-COMP result: unknown` would trigger the next conf.

This also relates to the previous comment about Goblint output visibility because the BenchExec tool info module parses errors into different statuses here: https://github.com/sosy-lab/benchexec/blob/4970b1aef889c9069e68989caa82591737ec098a/benchexec/tools/goblint.py#L89-L100.
We should still be able to distinguish those through the wrapper, e.g. distinguish various uncaught exceptions by their type in the BenchExec results table.